lib/cn/csched_sp3_work.c

```diff
@@ -688,54 +686,46 @@ sp3_work(
     }
 
     if (cn_node_isleaf(tn)) {
-
         switch (wtype) {
-            case wtype_leaf_size:
-                n_kvsets = sp3_work_leaf_size(spn, thresh, &mark, &action, &rule);
-                *qnum_out = SP3_QNUM_LEAFBIG;
-                break;
+        case wtype_leaf_size:
+            n_kvsets = sp3_work_leaf_size(spn, thresh, &mark, &action, &rule);
+            break;
 
-            case wtype_leaf_garbage:
-                n_kvsets = sp3_work_leaf_garbage(spn, thresh, &mark, &action, &rule);
-                *qnum_out = SP3_QNUM_LEAF;
-                break;
+        case wtype_leaf_garbage:
+            n_kvsets = sp3_work_leaf_garbage(spn, thresh, &mark, &action, &rule);
+            break;
 
-            case wtype_node_len:
-                n_kvsets = sp3_work_leaf_len(spn, thresh, &mark, &action, &rule, &bonus);
-                *qnum_out = SP3_QNUM_LEAF;
-                break;
+        case wtype_node_len:
+            n_kvsets = sp3_work_leaf_len(spn, thresh, &mark, &action, &rule, &bonus);
+            break;
 
-            case wtype_leaf_scatter:
-                n_kvsets = sp3_work_leaf_scatter(spn, thresh, &mark, &action, &rule, &bonus);
-                *qnum_out = SP3_QNUM_LSCAT;
-                break;
+        case wtype_leaf_scatter:
+            n_kvsets = sp3_work_leaf_scatter(spn, thresh, &mark, &action, &rule, &bonus);
+            break;
 
-            default:
-                ev_warn(1);
-                break;
+        default:
+            ev_warn(1);
+            break;
         }
     } else {
-        uint cmin, cmax;
+        uint cmin = thresh->rspill_kvsets_min;
+        uint cmax = thresh->rspill_kvsets_max;
 
         switch (wtype) {
-            case wtype_rspill:
-                cmin = thresh->rspill_kvsets_min;
-                cmax = thresh->rspill_kvsets_max;
-                *qnum_out = SP3_QNUM_ROOT;
-                break;
-            case wtype_node_len:
-                cmin = thresh->rspill_kvsets_min;
-                cmax = thresh->rspill_kvsets_max;
-                *qnum_out = SP3_QNUM_INTERN;
-                break;
-            case wtype_ispill:
+        case wtype_rspill:
+        case wtype_node_len:
+            break;
+
+        case wtype_ispill:
+            if (tn->tn_loc.node_level > 0) {
                 cmin = thresh->ispill_kvsets_min;
                 cmax = thresh->ispill_kvsets_max;
-                *qnum_out = SP3_QNUM_INTERN;
-                break;
-            default:
-                ev_warn(1);
-                goto locked_nowork;
+            }
```

Comment:
This change fixes the issue where we were spilling just one kvset from cn root.  As we discussed, there may be other/better ways to do this...